src/web/lib/components/AppHeader/index.js

```diff
@@ -59,7 +59,14 @@ export const AppHeader = props => {
     let currentImages = new Set();
 
     props.themeCustomBackgrounds
-      .forEach(({ name }) => currentImages.add(name));
+      .forEach(({ name }) => { 
+        currentImages.add(name);
+        const image = temporaryImageStore.get(name);
+        if (image) {
+          props.updateImage({ ...image, importing: true });
+          props.updateCustomBackground({ name });
```

Comment:
To help me with following your patch (also for reference in the future), could you briefly explain the purpose of those two lines? What happens if the line was not added?